sdk/containerregistry/azure-containerregistry/README.md

```diff
@@ -69,74 +69,218 @@ For more information please see [Container Registry Concepts](https://docs.micro
+
+Please note that each sample assumes there is a `CONTAINERREGISTRY_ENDPOINT` environment variable set to a string containing the `https://` prefix and the name of the login server, for example "https://myregistry.azurecr.io". Anonymous access samples are getting endpoint value from environment variable`CONTAINERREGISTRY_ANONREGISTRY_ENDPOINT`.
 
 ### List repositories
 
 Iterate through the collection of repositories in the registry.
 
-```python
-endpoint = os.environ["CONTAINERREGISTRY_ENDPOINT"]
+<!-- SNIPPET:sample_delete_tags.list_repository_names -->
 
-with ContainerRegistryClient(endpoint, DefaultAzureCredential(), audience="https://management.azure.com") as client:
+```python
+with ContainerRegistryClient(self.endpoint, self.credential) as client:
     # Iterate through all the repositories
     for repository_name in client.list_repository_names():
         print(repository_name)
 ```
 
+<!-- END SNIPPET -->
+
 ### List tags with anonymous access
 
 Iterate through the collection of tags in the repository with anonymous access.
 
-```python
-endpoint = os.environ["CONTAINERREGISTRY_ENDPOINT"]
+<!-- SNIPPET:sample_list_tags.list_tags_anonymous -->
 
-with ContainerRegistryClient(endpoint, DefaultAzureCredential(), audience="https://management.azure.com") as client:
-    manifest = client.get_manifest_properties("library/hello-world", "latest")
-    print(manifest.repository_name + ": ")
+```python
+endpoint = os.environ.get("CONTAINERREGISTRY_ANONREGISTRY_ENDPOINT")
+with ContainerRegistryClient(endpoint) as anon_client:
+    manifest = anon_client.get_manifest_properties("library/hello-world", "latest")
+    print(f"Tags of {manifest.repository_name}: ")
+    # Iterate through all the tags
     for tag in manifest.tags:
-        print(tag + "\n")
+        print(f"{tag}\n")
 ```
 
+<!-- END SNIPPET -->
+
 ### Set artifact properties
 
 Set properties of an artifact.
 
-```python
-endpoint = os.environ["CONTAINERREGISTRY_ENDPOINT"]
+<!-- SNIPPET:sample_set_image_properties.update_manifest_properties -->
 
-with ContainerRegistryClient(endpoint, DefaultAzureCredential(), audience="https://management.azure.com") as client:
-    # Set permissions on the v1 image's "latest" tag
+```python
+with ContainerRegistryClient(self.endpoint, self.credential) as client:
+    # Set permissions on image "library/hello-world:v1"
     client.update_manifest_properties(
         "library/hello-world",
-        "latest",
+        "v1",
         can_write=False,
         can_delete=False
     )
 ```
 
+<!-- END SNIPPET -->
+
 ### Delete images
 
 Delete images older than the first three in the repository.
 
-```python
-endpoint = os.environ["CONTAINERREGISTRY_ENDPOINT"]
+<!-- SNIPPET:sample_delete_images.delete_manifests -->
 
-with ContainerRegistryClient(endpoint, DefaultAzureCredential(), audience="https://management.azure.com") as client:
+```python
+with ContainerRegistryClient(self.endpoint, self.credential) as client:
```

Comment:
Please double check if we need to provide audience in the ctor?

Reply:
No need. - We'll use the default audience when not providing, and we have changed the default value to one that works across clouds.